core/coreapi/coreapi.go

```diff
@@ -112,16 +114,26 @@ type path struct {
 }
 
 // ParsePath parses path `p` using ipfspath parser, returns the parsed path.
-func ParsePath(p string) (coreiface.Path, error) {
+func (api *CoreAPI) ParsePath(ctx context.Context, p string, opts ...caopts.ParsePathOption) (coreiface.Path, error) {
```

Comment:
The second this *resolves* the path, it's no longer parsing it. Maybe, instead, expose a `ResolvePathString` method (or just keep the steps separate).